### ISSUE TITLE
BasicGestures unreachable

### ISSUE THREAD

**Issue by pelikhan** [lancaster-university/microbit-dal#109]
If `force` is 8g, isn't `GESTURE_3G` always going to trigger?

```
    if (force > MICROBIT_ACCELEROMETER_3G_THRESHOLD)
        return GESTURE_3G;

    if (force > MICROBIT_ACCELEROMETER_6G_THRESHOLD)
        return GESTURE_6G;

    if (force > MICROBIT_ACCELEROMETER_8G_THRESHOLD)
        return GESTURE_8G;
```

**Comment by finneyj** [comment]
true... These events haven't found many uses though - I wonder if we should just consider dropping them in the interest of simplicity.

Thoughts?

**Comment by dpgeorge** [comment]
These are now part of the public MicroPython API.  So removing them would potentially break scripts.

**Comment by finneyj** [comment]
Thanks @dpgeorge - we should keep them then.

**Comment by dpgeorge** [comment]
Sounds good!  But can they be fixed as well?

**Comment by finneyj** [comment]
:-) for sure

**Comment by finneyj** [comment]
I guess there's a question of semantics too. If we have say an impulse of 7G, should we fire both 3G and 6G event, or just the higher value?

**Comment by dpgeorge** [comment]
> If we have say an impulse of 7G, should we fire both 3G and 6G event, or just the higher value?

I'd say just the highest value.  Most uses of it would not want to process both 3G and 6G, so would need to go out of their way to discard the 3G if there was a 6G at the same time (and how does it know they were from the same event?).

**Comment by finneyj** [comment]
agree - good rationale.

**Comment by remay** [comment]
But won't it be odd if I register for 3G impulses and then shake the device hard and don't get triggered?

**Comment by finneyj** [comment]
An even better point! Thanks @remay.

Coming back to @dpgeorge's thoughts, I guess someone wanting only 8G events would register for that event specifically, so would need no filtering. If they register for all events, then they would expect to need to do some filtering... To me, this seems more intuitive than missing events.

**Comment by dpgeorge** [comment]
MicroPython registers for all events and exposes them as a list when they come in.  The list of pending events is cleared when the user reads that list.

**Comment by mmoskal** [comment]
I would say both 3G and 6G - this is because if someone shakes the device slightly differently, or you sample slightly differently, you will first get say 4.1G and then next sample 7.3G, instead of getting say 6.5G once.

**Comment by finneyj** [comment]
Yep. 

@dpgeorge - if we all document that an impulse will trigger any threshold event of that level or lower, then would this mitigate the issue any?

**Comment by dpgeorge** [comment]
Yes I guess that's fine (to post multiple events).  I agree that if you have an application that deals with such high forces they you need to handle the case of lower forces as well.

**Comment by finneyj** [comment]
Now fixed in master. Will be included in next tagged release (v2.0.0rc3)

I also took the opportunity to remove the damping for impulse events (3G/6G/8G and FREEFALL) so that they react more quickly...
